### ISSUE TITLE
AvoidNotShortCircuitOperatorsForBoolean reports multi catch case

### ISSUE THREAD

**Issue by george-007** [sevntu-checkstyle/sevntu.checkstyle#559]
/var/tmp $ javac Test.java

/var/tmp $ cat Test.java
```java
public class Test {
    public class InnerClass {
        public Runnable getRunnable() {
            return new Runnable() {
                @Override
                public void run() {
                }
                public boolean test() {
                    try {
                    } catch (IllegalArgumentException | NullPointerException e) {
                        return false;
                    }
                    return true;
                }
            };

        }
    }

}
```

/var/tmp $ cat config.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="AvoidNotShortCircuitOperatorsForBoolean"/>
    </module>
</module>
```
```
/var/tmp $ # java -classpath sevntu-checks-1.23.0.jar:checkstyle-7.3-all.jar com.puppycrawl.tools.checkstyle.Main  -c config.xml Test.java
```
```
Starting audit...
[ERROR] /root/checkstyle/Test.java:10:55: Short-circuit operator should be used instead of '|'. [AvoidNotShortCircuitOperatorsForBoolean]
Audit done.
Checkstyle ends with 1 errors.
```

The following code should not be reported as a violation:
```
} catch (IllegalArgumentException | NullPointerException e) {
```

**Comment by romani** [comment]
fix is merged

**Comment by toli** [comment]
This still seems to still fail for me on 
```
      try {
       // do stuff....
        docker.startContainer(docker.createContainer(containerConfig).id());
      } catch (InterruptedException | DockerException ex) {
        logger.info("exception while trying to start docker contaner: {}", ex.getMessage());
        return;
      }
```
with version `1.29.0` or `1.26.0` - i get an error 
```
[WARN] <path>File.java:598:37: Short-circuit operator should be used instead of '|'. [AvoidNotShortCircuitOperatorsForBoolean]
```

`DockerExcepiton` is deifined in https://github.com/spotify/docker-client library

**Comment by romani** [comment]
@toli , please create new issue with all details required in issue template.

**Comment by toli** [comment]
filed #659
